lib/fog/opennebula/requests/compute/image_pool.rb

```diff
@@ -18,10 +18,15 @@ def image_pool(filter = { })
               end
             end
           end
-          images.delete nil
           images
         end #def image_pool
       end #class Real
+
+      class Mock
```

Comment:
For mocks that don't do anything yet it is probably better to just leave these undefined (it should automatically return a nice error in this case).